Dockerfile

```diff
@@ -1,3 +1,15 @@
+FROM node:10 as frontend-builder
+
+COPY package.json package-lock.json webpack.config.js /frontend/
+COPY client/ /frontend/client
+WORKDIR /frontend
+
+COPY bin/bundle-extensions /frontend/bin/bundle-extensions
+RUN apt-get update -qq && apt-get install -y python-pip && pip install -U setuptools==23.1.0
+RUN npm run bundle
+
+RUN npm install && npm run build
```

Comment:
It's better to split `npm install` from the build, so if there was no change in packages, it will cache this layer.

Reply:
Also the COPY of the non package files should happen between install and build (for similar caching reasons).